relay-replays/src/recording.rs

```diff
@@ -562,4 +605,52 @@ mod tests {
         // NOTE: default scrubbers do not remove email address
         // assert!(scrubbed_result.contains("\"message\":\"[email]\""));
     }
+
+    #[test]
+    fn test_scrub_pii_key_based_default_config_without_static_rules() {
+        let payload = include_bytes!("../tests/fixtures/rrweb-request.json");
+
+        let mut transcoded = Vec::new();
+        let config = default_pii_config();
+
+        RecordingScrubber::from_configs(usize::MAX, Some(&config))
+            .scrub_replay(payload.as_slice(), &mut transcoded)
+            .unwrap();
+
+        let scrubbed_result = std::str::from_utf8(&transcoded).unwrap();
+        let scrubbed: serde_json::Value = serde_json::from_str(scrubbed_result).unwrap();
+
+        // Normal fields are not scrubbed:
+        assert_eq!(scrubbed[0]["data"]["payload"]["data"]["method"], "POST");
+
+        // `api_key` is caught by default scrubbing rules:
+        assert_eq!(
+            scrubbed[0]["data"]["payload"]["data"]["request"]["body"]["api_key"],
+            "[Filtered]"
```

Comment:
@billyvg this test illustrates that even without loading the new `STATIC_PII_CONFIG`, `api_key` is now filtered because the scrubber has now become aware of keys, and `api_key` is listed as a secret key in the default data scrubbing config:

https://github.com/getsentry/relay/blob/1cfd522d43ce55cc63606bf5d2874d59bcd45963/relay-general/src/pii/regexes.rs#L275-L277

This also means that from now on, these keys will be scrubbed anywhere in the payload (`passwd`, `auth`, `token`, ...). Is there a risk that this will break replays functionally?

Reply:
Oh ok, this answers my first comment if I'm understanding correctly. There is a risk if this is applied to all payloads (less risk if it's only where `type == 5`, but can still cause unwanted issues). Is it possible to only scrub for these keys within a path?

e.g. if in `STATIC_PII_CONFIG`:

```
let filter_paths = [vec![
        "data", "payload", "data", "request",
    ]];
```

I would want this assertion to be true:

```
assert_eq!(
            scrubbed[0]["data"]["payload"]["data"]["request"]["body"]["api_key"],
            "[Filtered]"
```

Reply:
@billyvg this is what I now implemented using `PII_FIELDS`. The selector is currently `vec!["data", "payload", "data"]`, but feel free to change that.